Change TxBuilder to look up keys from the config file

Geod24 opened the pull request:
This way we can deploy a TestNet where the keys are not known to everyone.

Fixes #82

kchulj reviewed (approved):
✅

kchulj commented:
Needs rebase, currently testing locally.

kchulj commented:
🟢 
Tested locally and works.

Geod24 commented:
We also need to add it when `None` is used no ?

kchulj commented:
> We also need to add it when `None` is used no ?

Yes good point.

hewison-chris commented on source/faucet/config.yaml:
```diff
@@ -4,6 +4,8 @@
 # When we deploy Faucet on the real TestNet, the keys used will not be well-known.
 keys:
+  # Genesis seed
+  - SDN7BBGE6Z6OQM3K4PACLTZUJ5QX4AY4QPDQ2JJ2JCFWCG2OIYYALIRY
```
Perhaps the comment can go at end of line to make it clear it is just that line.

hewison-chris replied:
Although maybe it is not possible.

kchulj replied:
Or another comment for the 6 keys below?

Geod24 commented:
Also need a few other fixes, hold on

Geod24 commented:
Updated with a few improvements:
1) We already had `getOwnedUTXO`, so I moved the filter call there;
2) If we need `Genesis` in the config file, we need it with `none`;
3) Moved the config file in its directory, like Agora, because code and example config should not mix;
4) Make the config file a little bit better w.r.t. real world usage by using A-Z.

hewison-chris reviewed (approved):
LGTM
